src/olympia/devhub/forms.py

```diff
@@ -557,23 +540,31 @@ class StandaloneValidationForm(AddonUploadForm):
             u'your own and only need it to be signed by Mozilla.'))
 
 
-class NewVersionForm(NewAddonForm):
+class NewVersionForm(AddonUploadForm):
+    supported_platforms = forms.TypedMultipleChoiceField(
+        choices=amo.SUPPORTED_PLATFORMS_CHOICES,
+        widget=forms.CheckboxSelectMultiple(attrs={'class': 'platform'}),
+        initial=[amo.PLATFORM_ALL.id],
+        coerce=int,
+        error_messages={'required': 'Need at least one platform.'}
+    )
+
     beta = forms.BooleanField(
         required=False,
         help_text=_lazy(u'A file with a version ending with '
                         u'a|alpha|b|beta|pre|rc and an optional number is '
                         u'detected as beta.'))
 
     def __init__(self, *args, **kw):
-        self.addon = kw.pop('addon')
+        self.addon = kw.pop('addon', None)
```

Comment:
`NewVersionForm` is used (now) both for a new version on an existing Addon and for a new Version on a new Addon (e.g. a new Addon).  In the latter case `addon` doesn't exist at this point.  Could pass in `addon=None` in the constructor but it seems less clear what's going on.